### DISCUSSION TITLE
want to press <Enter> to go a newline, but  autocomplete happens.

### DISCUSSION THREAD

**Discussion by ILOT-code** [ayamir/nvimdots#1116] [Q&A]
In insert model, when I need to press \<Enter\> to go to the next line, it will select the first completion option , even if I don't press \<Tab\> to select. And I had to press \<C-w\> to close the completion window and then press \<Enter\>

I'm wondering why even though I don't press \<Tab\> to enter the completion window, the \<Enter\> key still makes the completion selection instead of a new line.

When i read source code, in "cmp.lua", I find code:
```
		mapping = cmp.mapping.preset.insert({
			["<CR>"] = cmp.mapping.confirm({ select = true, behavior = cmp.ConfirmBehavior.Replace }),
```
If "select = true" was set, why does this behavior happen? Is there anything wrong? And how can i solove it?

here is a example:
![Snipaste_2023-12-28_19-57-04](https://github.com/ayamir/nvimdots/assets/82327624/e134f0cc-e9fd-4dd4-9d5a-a076bcec48d9)
Now I want to press Enter to write the main block, but if I press Enter, it will select the first completion:
![image](https://github.com/ayamir/nvimdots/assets/82327624/8440624a-22c8-441e-b8b8-96df803fb613)

**Comment by ayamir** [comment]
OK, I think set it to false is a better choice.

**Reply by ILOT-code** [reply]
From your words, I learned that I misunderstood the meaning of the "select=true" option. In fact, my original understanding was exactly the opposite, so I felt strange, haha.
Thanks.
